ala-name-matching-builder/src/main/java/au/org/ala/names/index/ALANameAnalyser.java

```diff
@@ -219,20 +219,20 @@ public NameKey analyse(@Nullable NomenclaturalClassifier code, String scientific
                 scientificName = (left + " " + right).trim();
             }
         }
+        try {
+            name = this.nameParser.parse(scientificName, (rankType == null || rankType == RankType.UNRANKED) ? null : rankType.getCbRank());
+            if (name != null) {
+                nameType = name.getType();
+                if (rankType == null && name.getRank() != null)
+                    rankType = RankType.getForCBRank(name.getRank());
+            }
+        } catch (UnparsableException ex) {
+            // Oh well, worth a try
```

Comment:
Just checking we might want to `log.error` this line for debugging prod issues ??

Reply:
This part of the pre-prod merge. If a name is unparsable, it is just treated as being "as-is" without extra information from parsing. Having said that, an log.info might be appropriate.

However, hang on before approving. I want to fix the "Genus B" problem, as well.